tests/validation/cattlevalidationtest/core/test_github.py

```diff
@@ -44,32 +45,122 @@ def test_github_token_has_access():
 
-    c = requests.post(BASE_URL + 'token', {'code': query['code']})
+    c = requests.post(BASE_URL + 'token', {'code': code})
+
+    return c.json()['jwt']
 
-    jwt = c.json()['jwt']
 
+@if_github
+def test_github_auth_config_unauth_user(github_request_token):
+    #   switch on auth
     requests.post(BASE_URL + 'githubconfig',
                   data=json.dumps({'enabled': 'true'}))
 
+#   do not set any auth headers
     no_auth = requests.get(URL)
 
+#   test that auth is switched on
     assert no_auth.status_code == 401
 
+    jwt = github_request_token
+#   switch it back
+    requests.post(BASE_URL + 'githubconfig',
+                  headers={'Authorization': 'Bearer ' + jwt},
+                  data=json.dumps({'enabled': 'false'}))
+
+
+@if_github
+def test_github_auth_config_invalid_user(github_request_token):
+    #   switch on auth
+    requests.post(BASE_URL + 'githubconfig',
+                  data=json.dumps({'enabled': 'true'}))
+
+#   set invalid auth headers
     bad_auth = requests.get(URL,
                             headers={'Authorization':
                                      'Bearer some_random_string'})
 
+#   test that user does not have access
     assert bad_auth.status_code == 401
 
+    jwt = github_request_token
+#   switch it back
+    requests.post(BASE_URL + 'githubconfig',
+                  headers={'Authorization': 'Bearer ' + jwt},
+                  data=json.dumps({'enabled': 'false'}))
+
+
+@if_github
+def test_github_auth_config_valid_user(github_request_token):
+    #   switch on auth
+    requests.post(BASE_URL + 'githubconfig',
+                  data=json.dumps({'enabled': 'true'}))
+
+    jwt = github_request_token
+
+#   set valid auth headers
     schemas = requests.get(URL, headers={'Authorization': 'Bearer ' + jwt})
 
+#   test that user has access
     assert schemas.status_code == 200
 
+#   switch it back
     requests.post(BASE_URL + 'githubconfig',
                   headers={'Authorization': 'Bearer ' + jwt},
                   data=json.dumps({'enabled': 'false'}))
 
-    driver.quit()
+
+@if_github
+def test_github_auth_config_api_whitelist_users(github_request_token):
+    #   set whitelisted users
+    requests.post(BASE_URL + 'githubconfig',
+                  data=json.dumps({'allowedUsers':
+                                  ['sidharthamani', 'ibuildthecloud']}))
+
+#   test that these users were whitelisted
+    r = requests.get(BASE_URL + 'githubconfig')
+
+    users = r.json()['data'][0]['allowedUsers']
+
+    user_count = 2
+
+    assert len(users) == 2
+
+    for user in users:
+        if user == 'sidharthamani' or user == 'ibuildthecloud':
+            user_count = user_count - 1
+
+    assert user_count == 0
+
+
+@if_github
+def test_github_auth_config_api_whitelist_orgs(github_request_token):
+    #   set whitelisted users
+    requests.post(BASE_URL + 'githubconfig',
+                  data=json.dumps({'allowedOrganizations': ['rancherio']}))
```

Comment:
Similar thing, this should test that if an org API_AUTH_GITHUB_TEST_USER belongs to auth succeeds, and if they are not that it fails.